.github/workflows/build.yml

```diff
@@ -9,46 +9,62 @@ on:
   pull_request:
 jobs:
   build:
-    name: Build images
+    name: Build binaries
     runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    # Build binaries
+    - name: Run ci
+      run: make ci
+
+    - name: Upload binaries
+      uses: actions/upload-artifact@v4
+      with:
+        name: binaries_artifact
+        path: ./bin/*
+
+  build_push_image:
+    name: Build and push images
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ startsWith(github.ref, 'refs/heads/') || startsWith(github.ref, 'refs/tags/') }}
     permissions:
       contents: read
       id-token: write # for reading credential https://github.com/rancher-eio/read-vault-secrets
-    container:
-      image: rancher/dapper:v0.6.0
     steps:
-    # Git is not in Dapper container image. Add it manually for dirty check.
-    - name: Add Git
-      run: apk add -U git
     - name: Checkout code
       uses: actions/checkout@v4
-    - name: Declare branch
-      run: |
-        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_ENV"
 
-    # For multi-platform support
-    - name: Set up QEMU
-      uses: docker/setup-qemu-action@v3
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v3
+    - name: Download binaries
+      uses: actions/download-artifact@v4
+      with:
+        name: binaries_artifact
+        path: ./bin/
```

Comment:
Can you help double-check if the binaries are executable? It seems to be packed with zip.

Reply:
Yeah, we miss it. I add `chmod +x` before copying files to `package` folder. Thank you.